model_compression_toolkit/target_platform_capabilities/schema/v1.py

```
@@ -203,64 +219,92 @@ def clone_and_edit(self, attr_to_edit: Dict[str, Dict[str, Any]] = {}, **kwargs)
-        for cfg in self.quantization_configurations:
-            if not isinstance(cfg, OpQuantizationConfig):
-                Logger.critical(
-                    f"Each option must be an instance of 'OpQuantizationConfig', but found an object of type: {type(cfg)}.") # pragma: no cover
+                f"'quantization_configurations' must be a list or tuple, but received: {type(quantization_configurations)}.") # pragma: no cover
 
-        # Handle base_config
-        if len(self.quantization_configurations) > 1:
-            if self.base_config is None:
-                Logger.critical(f"For multiple configurations, a 'base_config' is required for non-mixed-precision optimization.") # pragma: no cover
-            if not any(self.base_config == cfg for cfg in self.quantization_configurations):
-                Logger.critical(f"'base_config' must be included in the quantization config options.") # pragma: no cover
-        elif len(self.quantization_configurations) == 1:
-            if self.base_config is None:
-                object.__setattr__(self, 'base_config', self.quantization_configurations[0])
-            elif self.base_config != self.quantization_configurations[0]:
+        if num_configs > 1:
+            if base_config is None:
                 Logger.critical(
-                    "'base_config' should be the same as the sole item in 'quantization_configurations'.") # pragma: no cover
+                    "For multiple configurations, a 'base_config' is required for non-mixed-precision optimization.") # pragma: no cover
+            if base_config not in quantization_configurations:
+                Logger.critical("'base_config' must be included in the quantization config options.") # pragma: no cover
+        elif num_configs == 1:
+            if base_config is None:
+                # Automatically set base_config to the sole configuration
```

was this the previous behavior? if not, I think it is better to throw an error in this case

---

This is previous behavior